.jenkins/Jenkinsfile

```diff
@@ -250,227 +198,149 @@ def isCanonical() {
+        
+        cleanAgent(".build/docker/run-tests.sh")
       }
     }
   }
 }
 
+def fetchSource(stage, arch, jdk) {
+    if ("jar" == stage) {
+        checkout changelog: false, scm: scmGit(branches: [[name: params.branch]], extensions: [cloneOption(depth: 1, noTags: true, reference: '', shallow: true)], userRemoteConfigs: [[url: params.repository]])
+    } else {
+        unstash name: "${arch}_${jdk}"
+    }
+}
+
 /**
  * Saves JUnit results, fails if no reports are found
  * Copies (single-file unified) junit results and logfiles to nightlies
  * Cleans the agent (docker prune) afterwards
  * Stops the pipeline on any failure
  **/
-def testJob(test_script, test_type, splits=1) {
-  // we don't actually need the test_script and test_type, but they are useful explicit for readability and intuitive feedback in the declarative section.
-  assert STAGE_NAME.equals(test_type) : "Stage's name expected to match test_type (" + STAGE_NAME + " != " + test_type + ")"
-  assert ".build/docker/run-tests.sh".equals(test_script) : "testJob is hardcoded to '.build/docker/run-tests.sh', found " + test_script
-
-  if (! isStageIncluded()) {
-    echo "Skipping ${STAGE_NAME}"
-    return;
-  }
-
-  // XXX – the following matrix over arrays can be better done with multi-arrays and recursion
-
-  def archs = ['cassandra']
-  def jdks = STAGE_NAME.contains("dtest-upgrade") || "simulator-dtest".equals(STAGE_NAME) ? ["${javaVersionDefault}"] : "${javaVersionsSupported}".split(/,/, -1)
-  def pythons = "cqlshX".equals(STAGE_NAME) ? ['3.7', '3.8', '3.11'] : ['3.7']
-  def cythons = "cqlshX".equals(STAGE_NAME) ? ['no', 'yes'] : ['no']
-
-  def tests_per_arch = [:]
-  for (a in archs) {
-    def arch = a
-    tests_per_arch[arch] = {
-      def tests_per_jdk = [:]
-      for (j in jdks) {
-        def jdk = j
-        tests_per_jdk[jdk] = {
-          def tests_per_python = [:]
-          for (p in pythons) {
-            def python = p
-            tests_per_python[python] = {
-              def tests_per_cython = [:]
-              for (c in cythons) {
-                def cython = c
-                tests_per_cython[cython] = {
-                  def tests_per_split = [:]
-                  for (int s = 1; s <= splits; s++) {
-                    def split = s
-                    tests_per_split[split] = { _testAxis(test_script, test_type, arch, jdk, python, cython, "${split}/${splits}") }
-                  }
-                  parallel(tests_per_split)
-                }
-              }
-              parallel(tests_per_cython)
-            }
-          }
-          parallel(tests_per_python)
-        }
-      }
-      parallel(tests_per_jdk)
-    }
-  }
-  parallel(tests_per_arch)
-}
+// def testJob(test_type, splits=1) {
+//   def jdks = STAGE_NAME.contains("dtest-upgrade") || "simulator-dtest".equals(STAGE_NAME) ? ["${javaVersionDefault}"] : "${javaVersionsSupported}".split(/,/, -1)
```

Comment:
Could you please explain me what's going on here? It's not implemented in my version yet.

Reply:
I don't know what that commented out lines is.  just delete.

Reply:
done in my commit